lib/provisioner/plugin/utils.rb

```diff
@@ -48,6 +48,14 @@ def to_json(*a)
         end
       end
 
+      # Gets a host's SSH host key
+      def ssh_keyscan(host, type = 'rsa')
+        keytype = type == 'dsa' ? 'dss' : type
+        # TODO: find a way to do this in Ruby
+        key = `ssh-keyscan -t #{type} #{host} 2>&1 | grep #{keytype}`.split(' ')
```

Comment:
It ships with OpenSSH, so I assume it'll be present on any machine that we support. Should I still check/error? I didn't think it was necessary, but I'll do it if you think it should be done.